feature/text_based.py

```diff
@@ -0,0 +1,724 @@
+
+        # Range of rows
+        self.rows = range(self.matrix.shape[0])
+
+        # Range of columns
+        self.columns = range(self.matrix.shape[1])
+
+    @staticmethod
+    def _verify_input(cost, matrix):
+        assert (cost is not None), "Data Error: cost cannot be none"
+        assert (matrix is not None), "Data Error: matrix cannot be none"
+        assert (matrix.ndim == 2), "Data Error: matrix should 2D"
+        assert (len(cost) == matrix.shape[1]), \
+            "Data Error: cost vector size should match the number of columns " + \
+            str(len(cost)) + " vs. " + str(matrix.shape[1])
+
+
+class ContentSelector:
+    """ Content Selector.
+
+    This class contains a content selection algorithm (max_cover) that maximizes the diversity among the selected
+    content, while minimizing the size of the selection that can cover all (or a maximum subset) of a set of
+    predefined labels belonging to the content. The approach is closely related to set covering and is solved using a
+    multi-level optimization framework.
+
+    The class also includes baseline implementations that selects content based on a KMeans clustering of the content
+    features (kmeans) or simple random selection (random).
+
+    Attributes
+    ----------
+    selection_size: int
+        Maximum number of content to select.
+    seed: int
+        Random seed.
+    trials: int
+        Number of trials to run for random functions to estimate average coverage.
+    verbose: bool
+        Print intermediate results if True.
+    matrix: np.ndarray
+        Coverage matrix where each row corresponds to all predefined labels for each category and the columns
+        correspond to the content.
+    features: np.ndarray
+        Numeric featurization of content.
+
+    """
+
+    def __init__(self,  selection_size: int, seed: int = 123546, trials: int = 100, verbose: int = 0):
+        """ContentSelector
+
+       Creates a content selector object.
+
+       Parameters
+       ----------
+        selection_size: int
+            Maximum number of content to select.
+        seed: int, default=123456
+            Random seed.
+        trials: int, default=100
+            Number of trials to run for random functions to estimate average coverage.
+        verbose: int, default=0
+            Print intermediate results if 1.
+
+        """
+
+        # Validate arguments
+        self._validate_args(selection_size)
+
+        # Set input arguments
+        self.selection_size = selection_size
+        self.seed = seed
+        self.trials = trials
+        self.verbose = verbose == 1
+
+
+        # Initialize class variables
+        self.matrix = None
+        self.features = None
+        self._num_rows = None
+        self._num_cols = None
+
+    def run_content_selection(self,
+                              input_df: pd.DataFrame,
+                              categories: List[int], featurization_method: TextWiser, method: str = "random",
```

Comment:
move featurization_method to next line

Reply:
shall we replace method with "optimization_method" so that it is consistent with the outer TextBased class? Also, move it to other line